src/sagemaker/multidatamodel.py

```diff
@@ -0,0 +1,308 @@
+        in ``MultiDataModel`` constructor will be used. If ``self.predictor_cls`` is not
+        None, this method returns a the result of invoking ``self.predictor_cls`` on
+        the created endpoint name.
+
+        The name of the created model is accessible in the ``name`` field of
+        this ``Model`` after deploy returns
+
+        The name of the created endpoint is accessible in the
+        ``endpoint_name`` field of this ``Model`` after deploy returns.
+
+        Args:
+            initial_instance_count (int): The initial number of instances to run
+                in the ``Endpoint`` created from this ``Model``.
+            instance_type (str): The EC2 instance type to deploy this Model to.
+                For example, 'ml.p2.xlarge', or 'local' for local mode.
+            accelerator_type (str): Type of Elastic Inference accelerator to
+                deploy this model for model loading and inference, for example,
+                'ml.eia1.medium'. If not specified, no Elastic Inference
+                accelerator will be attached to the endpoint. For more
+                information:
+                https://docs.aws.amazon.com/sagemaker/latest/dg/ei.html
+            endpoint_name (str): The name of the endpoint to create (default:
+                None). If not specified, a unique endpoint name will be created.
+            update_endpoint (bool): Flag to update the model in an existing
+                Amazon SageMaker endpoint. If True, this will deploy a new
+                EndpointConfig to an already existing endpoint and delete
+                resources corresponding to the previous EndpointConfig. If
+                False, a new endpoint will be created. Default: False
+            tags (List[dict[str, str]]): The list of tags to attach to this
+                specific endpoint.
+            kms_key (str): The ARN of the KMS key that is used to encrypt the
+                data on the storage volume attached to the instance hosting the
+                endpoint.
+            wait (bool): Whether the call should wait until the deployment of
+                this model completes (default: True).
+            data_capture_config (sagemaker.model_monitor.DataCaptureConfig): Specifies
+                configuration related to Endpoint data capture for use with
+                Amazon SageMaker Model Monitoring. Default: None.
+
+        Returns:
+            callable[string, sagemaker.session.Session] or None: Invocation of
+                ``self.predictor_cls`` on the created endpoint name,
+                if ``self.predictor_cls``
+                is not None. Otherwise, return None.
+        """
+        # Set model specific parameters
+        if self.model:
+            enable_network_isolation = self.model.enable_network_isolation()
+            role = self.model.role
+            vpc_config = self.model.vpc_config
+            predictor = self.model.predictor_cls
```

Comment:
could this not simply be:

self.model.deploy()

else do super.deploy() ?

Reply:
I tried `self.model.deploy` but realized that since we set the values of `model_data_url` and `container_mode` parameters in `MultiDataModel`'s `prepare_container_def()` method, we'd need to replicate the logic from `Model`'s `deploy()` here, else it ends up creating a container with `single_model` mode and incorrect `model_data_url`